metrics/metrics_collector.go

```diff
@@ -28,37 +37,43 @@ func newMetricCollector() metricCollector {
 
 func (mc *metricCollector) updateCounter() {
 	for _, c := range mc.counters {
-		c.Add(rand.Float64())
+		for i := 0; i < mc.datapointCount; i++ {
+			labels := datapointLabels(i, mc.labelKeys, mc.labelValues)
+			c.With(labels).Add(rand.Float64())
+		}
 	}
-
 }
 
 func (mc *metricCollector) updateGauge() {
 	for _, c := range mc.gauges {
-		c.Add(rand.Float64())
+		for i := 0; i < mc.datapointCount; i++ {
+			labels := datapointLabels(i, mc.labelKeys, mc.labelValues)
+			c.With(labels).Set(rand.Float64())
+		}
 	}
 }
 
 func (mc *metricCollector) updateHistogram() {
 	for idx := 0; idx < len(mc.histograms); idx++ {
-		lowerBound := math.Mod(rand.Float64(), 1)
-		increment := math.Mod(rand.Float64(), 0.05)
-		for i := lowerBound; i < 1; i += increment {
-			mc.histograms[idx].Observe(i)
-
+		for i := 0; i < mc.datapointCount; i++ {
+			lowerBound := math.Mod(rand.Float64(), 1)
+			increment := math.Mod(rand.Float64(), 0.05)
+			labels := datapointLabels(i, mc.labelKeys, mc.labelValues)
+			for j := lowerBound; j < 1; j += increment {
+				mc.histograms[idx].With(labels).Observe(j)
+			}
```

Comment:
Shouldn't this also use a random distribution rather than incremental observations.

Reply:
@Aneurysm9 thanks. I have updated.